### PR TITLE
Convert remaining planning tests to typescript

### PR DESCRIPTION
**Pull request by lindner**

### CODE REVIEW

**Comment by lindner** [comment]
NOTE: still some tests to adjust using a new FakeVariable class..  Wanted to get this out for early feedback on approach.

Thanks!

File: src/runtime/plan/planning-result.ts [hunk 1]
```diff
@@ -194,7 +194,7 @@ export class PlanningResult {
            oldSuggestions.every(suggestion => newSuggestions.find(newSuggestion => suggestion.isEquivalent(newSuggestion)));
   }
 
-  fromLiteral({suggestions, generations, lastUpdated}) {
+  fromLiteral({suggestions, generations, lastUpdated}: {suggestions, generations?, lastUpdated?: Date}) {
```
[the remark this answers is not in the document]
Comment (lindner) [reply src/runtime/plan/planning-result.ts]: yes, it is.  A better way is to create a named type.  Will consider that.

File: src/runtime/test/plan/plan-producer-test.ts [hunk 2]
```diff
@@ -153,36 +161,64 @@ class TestPlanProducer extends PlanProducer {
...
+      fakeSearchStore.values = [{arc: arcKey, search}];
       return this.onSearchChanged();
     }
   }
 
-  function init() {
-    const searchStore = {on: () => {}};
-    searchStore.get = () => searchStore.values;
-    const producer = new TestSearchPlanProducer(searchStore);
+  class FakeVariable extends StorageProviderBase implements VariableStorageProvider {
```
[the remark this answers is not in the document]
Comment (lindner) [reply src/runtime/test/plan/plan-producer-test.ts]: I was attempting to be minimal and use the settable value that the current code uses.  I'm all for using Volatile and will adjust for that.

**Comment by lindner** [comment]
I will hold off on this until your suggestion merge PR is submitted.

File: src/runtime/plan/plan-consumer.ts [hunk 3]
```diff
@@ -46,7 +46,7 @@ export class PlanConsumer {
   registerSuggestionsChangedCallback(callback) { this.suggestionsChangeCallbacks.push(callback); }
   registerVisibleSuggestionsChangedCallback(callback) { this.visibleSuggestionsChangeCallbacks.push(callback); }
 
-  setSuggestFilter(showAll, search) {
+  setSuggestFilter(showAll?: boolean, search?: string) {
```
Comment (mmandlis) [inline src/runtime/plan/plan-consumer.ts]: actually, i don't `showAll` is not optional, please, remove the ?

Comment (lindner) [reply src/runtime/plan/plan-consumer.ts]: done

File: src/runtime/plan/planning-result.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lindner) [reply src/runtime/plan/planning-result.ts]: going to leave this as is for now to keep changes minimal.

**Pull request merged**